src/keydra/providers/qualys.py

```diff
@@ -0,0 +1,106 @@
+import boto3
+import json
+import validators
+
+from keydra.clients.qualys import QualysClient
+from keydra.clients.aws.secretsmanager import SecretsManagerClient
+
+from keydra.providers.base import BaseProvider
+from keydra.providers.base import exponential_backoff_retry
+
+from keydra.exceptions import DistributionException
+from keydra.exceptions import RotationException
+
+from keydra.logging import get_logger
+
+LOGGER = get_logger()
+
+
+class Client(BaseProvider):
+    def __init__(self, session=None, credentials=None, region_name=None):
+
+        if session is None:
+            session = boto3.session.Session()
+
+        smclient = SecretsManagerClient(
```

Comment:
Oh, so we're tightly coupling Qualys provider to AWS SM.. Could we perhaps make it config-driven, e.g. add 
```
config:
  rotatewith:
    key: keydra/qualys/secondary
    provider: secretsmanager
```